tests/e2e/scale/test_cephfs_many_files.py

```diff
@@ -0,0 +1,238 @@
+        logging.info("Teardown complete")
+
+
+@pytest.fixture(scope='session')
+def million_file_cephfs(request):
+    million_file_cephfs = MillionFilesOnCephfs()
+
+    def teardown():
+        million_file_cephfs.cleanup()
+    request.addfinalizer(teardown)
+
+
+@scale
+@ignore_leftovers
+@pytest.mark.parametrize(
+    argnames=["resource_to_delete", "build_temp"],
+    argvalues=[
+        pytest.param(
+            *['mgr', True],
+        ),
+        pytest.param(
+            *['mon', False],
+        ),
+        pytest.param(
+            *['osd', False],
+        ),
+        pytest.param(
+            *['mds', False],
+        ),
+    ]
+)
+class TestMillionCephfsFiles(E2ETest):
+    """
+    Million cephfs files tester.
+    """
+    def test_scale_million_cephfs_files(
+        self,
+        million_file_cephfs,
+        resource_to_delete,
+        build_temp
+    ):
+        """
+        Add a million files to the ceph filesystem
+        Delete each instance of the parametrized ceph pod
+        Once the ceph cluster is healthy, verify that no files were lost
+
+        args:
+            million_file_cephfs -- fixture
+            resource_to_delete (str) -- resource deleted for each testcase
+            build_temp -- Add one million files, if True.
+        """
+        if build_temp:
+            self.sample_list = add_million_files()
+        else:
+            with open(JSON_FNAME, 'r') as fd:
+                self.sample_list = json.load(fd)
+        proc = subprocess.Popen(
+            f'oc -n {DEFAULT_NS} rsh {POD_NAME} df | grep {TARGET_DIR}',
+            shell=True,
+            stdout=subprocess.PIPE
+        )
+        dfoutput = proc.communicate()[0]
+        logging.info(f"Df results on ceph pod - {dfoutput}")
+        if resource_to_delete in ['mgr', 'mon', 'osd', 'mds']:
+            logging.info(f"Testing respin of {resource_to_delete}")
+            disruption = disruption_helpers.Disruptions()
+            disruption.set_resource(resource=resource_to_delete)
+            no_of_resources = disruption.resource_count
+            for i in range(0, no_of_resources):
+                disruption.delete_resource(resource_id=i)
+            retry(
+                exceptions.CephHealthException,
+                tries=60, delay=5, backoff=1
+            )(utils.ceph_health_check)()
+        logging.info("Verifying that the file count has not changed")
+        lsdata = subprocess.Popen([
+            'oc',
+            '-n',
+            DEFAULT_NS,
+            'exec',
+            POD_NAME,
+            '--',
+            'ls',
+            f'{TARGET_DIR}/x'
+        ], stdout=subprocess.PIPE)
+        output = subprocess.check_output([
+            'wc',
+            '-w'
+        ], stdin=lsdata.stdout)
+        lsdata.wait()
+        file_count = int(output)
+        assert file_count == TFILES
+        logging.info("Testing renaming of files")
+        nsample = []
+        for sample in self.sample_list:
```

Comment:
What is the purpose of this part of the code?

Reply:
This code tests renaming of files after the ceph cluster is healthy again.   I can probably choose to rename a separate set of files rather than keep track of name changes with that awkward json file.  I